requests/PermissionRequest.go

```diff
@@ -0,0 +1,65 @@
+package requests
+
+// PermissionRequest structure to change permission
+type PermissionRequest struct {
+	Permissions [][]int
+}
+
+// PermissionRequestBuilder structure to create permission permissionRequest
+type PermissionRequestBuilder struct {
+	permissionRequest PermissionRequest
+}
+
+// PermissionGroup type
+type PermissionGroup int
+
+const (
+	// User of PermissionGroup
+	User PermissionGroup = iota
+	// Group of PermissionGroup
+	Group
+	// Other of PermissionGroup
+	Other
+)
+
+// PermissionType type
+type PermissionType int
+
+const (
+	// Use of PermissionType
+	Use PermissionType = iota
+	// Manage of PermissionGroup
+	Manage
+	// Admin of PermissionGroup
+	Admin
+)
+
+// Allow to change permission
+const Allow = 1
+
+// Deny to change permission
+const Deny = 0
+
+// NoChange to set default permission permissionRequest
+const NoChange = -1
+
+// CreatePermissionRequestBuilder constructs PermissionRequestBuilder of PermissionRequest
+func CreatePermissionRequestBuilder() *PermissionRequestBuilder {
+	return &PermissionRequestBuilder{
+		PermissionRequest{[][]int{{NoChange, NoChange, NoChange}, {NoChange, NoChange, NoChange}, {NoChange, NoChange, NoChange}}}}
+}
+
+// Allow to change permission
+func (b *PermissionRequestBuilder) Allow(pg PermissionGroup, pt PermissionType) {
```

Comment:
The builder pattern was alright I just wanted those methods to return a pointer not a copy. Or is it a problem?

Reply:
This method is supposed to modify the state of the builder, so I believe acting on pointer is more appropriate in this case. For chaining, we also want to reuse the same builder, so returning a pointer in `Allow` and `Deny` methods should be good aswell. However, the `Build` methods should both act on a value (as they don't modify the state of the builder) and return a value. I will try to leave some useful comments about this in other builders, too :smile:

Reply:
Ok, this makes sense and I agree. But what exactly are we gaining by returning a value instead of a pointer from a constructor (eg. `CreateOwnershipRequestBuilder`)?